Issue when parsing struct iwl_mvm_reorder_buf_entry on drivers/net/wireless/intel/iwlwifi/mvm/mld-sta.c

The problem happens when extracting function iwl_mvm_mld_rm_sta_id from file drivers/net/wireless/intel/iwlwifi/mvm/mld-sta.c.

The problematic struct is definition is 

```c
struct iwl_mvm_reorder_buf_entry {                                                                                    
        struct sk_buff_head frames;                                                                                   
}                                                                                                                     
#ifndef __CHECKER__                                                                                                   
/* sparse doesn't like this construct: "bad integer constant expression" */                                           
__aligned(roundup_pow_of_two(sizeof(struct sk_buff_head)))                                                            
#endif                                                                                                                
;  
```

clang-extract errors out on ClosurePass:

```
drivers/net/wireless/intel/iwlwifi/mvm/mld-sta.c:2507:59: error: expected ';' after struct                            
 2507 | __aligned(roundup_pow_of_two(sizeof(struct sk_buff_head)))                                                    
      |                                                           ^                                                   
      |                                                           ;                                                   
                                                                                                                      
Error on pass: ClosurePass 
```

--- a/libcextract/PrettyPrint.cpp
+++ b/libcextract/PrettyPrint.cpp
@@ -424,6 +424,20 @@ SourceLocation PrettyPrint::Get_Expanded_Loc(Decl *decl)
         if (tok->is(semicolon)) {
           /* Found the ';', now go back a character to *not* include the ';'.  */
           furthest = head.getLocWithOffset(-1);
+
+          /* There is a bizarre case where a decl have an #endif at the end of
+             it, just before the ';' (see small/attr-14.c).  Try to workaround
+             this case by emiting the ';' that was there, which will result in
+             two ';' being print.  This is visually not cute, but at least
+             won't cause the compiler to cry.  */
+          StringRef endstr = PrettyPrint::Get_Source_Text({furthest, furthest});
+          if (const char *data_str = endstr.data()) {
+            /* Check for newline.  */
+            if (*data_str != '\0' && *data_str == '\n') {
+              /* Set 'furthest' back to where it was.  */
+              furthest = furthest.getLocWithOffset(1);
+            }
+          }
           break;
         }
       }

--- a/testsuite/small/attr-14.c
+++ b/testsuite/small/attr-14.c
@@ -1,0 +1,17 @@
+/* { dg-options "-DCE_EXTRACT_FUNCTIONS=f -DCE_NO_EXTERNALIZATION" }*/
+
+struct iwl_mvm_reorder_buf_entry {
+  int frames;
+}
+#ifndef __CHECKER__
+/* sparse doesn't like this construct: "bad integer constant expression" */
+__attribute__((aligned(32)))
+#endif
+;
+
+int f(struct iwl_mvm_reorder_buf_entry x)
+{
+  return x.frames;
+}
+
+/* { dg-final { scan-tree-dump "__attribute__\(\(aligned\(32\)\)\)" } } */

